crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2611,6 +2610,49 @@ impl<'db> TypeInferenceBuilder<'db> {
                     op,
                     Type::IntLiteral(i64::from(b)),
                 ),
+
+            (Type::StringLiteral(salsa_s1), Type::StringLiteral(salsa_s2)) => {
+                let s1 = salsa_s1.value(self.db);
+                let s2 = salsa_s2.value(self.db);
+                match op {
+                    ast::CmpOp::Eq => Some(Type::BooleanLiteral(s1 == s2)),
+                    ast::CmpOp::NotEq => Some(Type::BooleanLiteral(s1 != s2)),
+                    ast::CmpOp::Lt => Some(Type::BooleanLiteral(s1 < s2)),
+                    ast::CmpOp::LtE => Some(Type::BooleanLiteral(s1 <= s2)),
+                    ast::CmpOp::Gt => Some(Type::BooleanLiteral(s1 > s2)),
+                    ast::CmpOp::GtE => Some(Type::BooleanLiteral(s1 >= s2)),
+                    ast::CmpOp::In => Some(Type::BooleanLiteral(s2.contains(s1.as_ref()))),
+                    ast::CmpOp::NotIn => Some(Type::BooleanLiteral(!s2.contains(s1.as_ref()))),
+                    ast::CmpOp::Is => {
+                        if s1 == s2 {
+                            Some(KnownClass::Bool.to_instance(self.db))
+                        } else {
+                            Some(Type::BooleanLiteral(false))
+                        }
+                    }
+                    ast::CmpOp::IsNot => {
+                        if s1 == s2 {
+                            Some(KnownClass::Bool.to_instance(self.db))
+                        } else {
+                            Some(Type::BooleanLiteral(true))
+                        }
+                    }
+                }
+            }
+            (Type::StringLiteral(_), _) => {
+                self.infer_binary_type_comparison(Type::builtin_str_instance(self.db), op, right)
+            }
+            (_, Type::StringLiteral(_)) => {
+                self.infer_binary_type_comparison(left, op, Type::builtin_str_instance(self.db))
+            }
+
+            (Type::LiteralString, _) => {
+                self.infer_binary_type_comparison(Type::builtin_str_instance(self.db), op, right)
+            }
+            (_, Type::LiteralString) => {
+                self.infer_binary_type_comparison(left, op, Type::builtin_str_instance(self.db))
```

Comment:
```suggestion
                self.infer_binary_type_comparison(KnownClass::Str.to_instance(self.db), op, right)
            }
            (_, Type::StringLiteral(_)) => {
                self.infer_binary_type_comparison(left, op, KnownClass::Str.to_instance(self.db))
            }

            (Type::LiteralString, _) => {
                self.infer_binary_type_comparison(KnownClass::Str.to_instance(self.db), op, right)
            }
            (_, Type::LiteralString) => {
                self.infer_binary_type_comparison(left, op, KnownClass::Str.to_instance(self.db))
```

Reply:
Saw around the same time, a fixup's already pushed